src/pools/gyro2Pool/gyro2Pool.ts

```diff
@@ -160,20 +161,42 @@ export class Gyro2Pool implements PoolBase {
     }
 
     getLimitAmountSwap(
-        poolPairData: PoolPairBase,
+        poolPairData: Gyro2PoolPairData,
         swapType: SwapTypes
     ): OldBigNumber {
         if (swapType === SwapTypes.SwapExactIn) {
+            const balances = [poolPairData.balanceIn, poolPairData.balanceOut];
+            const normalizedBalances = _normalizeBalances(
+                balances,
+                poolPairData.decimalsIn,
+                poolPairData.decimalsOut
+            );
+            const invariant = _calculateInvariant(
+                normalizedBalances,
+                poolPairData.sqrtAlpha,
+                poolPairData.sqrtBeta
+            );
+            const maxAmountInAssetInPool = mulDown(
+                invariant,
+                poolPairData.sqrtBeta.sub(poolPairData.sqrtAlpha)
+            ); // y+ = L * (sqrtBeta - sqrtAlpha)
+            const limitAmountIn = maxAmountInAssetInPool.sub(
+                normalizedBalances[0]
+            );
+            const limitAmountInPlusSwapFee = divDown(
+                limitAmountIn,
+                ONE.sub(poolPairData.swapFee)
+            );
             return bnum(
                 formatFixed(
-                    mulDown(poolPairData.balanceIn, this.MAX_IN_RATIO),
-                    poolPairData.decimalsIn
+                    mulDown(limitAmountInPlusSwapFee, SWAP_LIMIT_FACTOR),
+                    18
```

Comment:
Is it intended to change the number of decimals from `poolPairData.decimalsIn` to 18 here? You didn't do this for the else branch.